### PR TITLE
Added tokenize keyword arguments to feature extraction pipeline

### LINKED ISSUE #19374: Feature extraction pipeline not consider parameters
### System Info

    - transformers==4.22.2
    - python==3.9.2
    - Ubuntu 22.04

### Who can help?

@Narsil 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

```python
import torch
from torch.utils.data import Dataset

from transformers import AutoModel, pipeline
from transformers import AutoTokenizer

model_name="anferico/bert-for-patents"

text = ["this is a pan"]
device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
model = AutoModel.from_pretrained(model_name).to(device)
tokenizer = AutoTokenizer.from_pretrained(model_name, do_lower_case=True, model_max_length=512)
pipe_ = pipeline('feature-extraction', model=model, tokenizer=tokenizer, device=torch.cuda.current_device())
p = pipe_(text, padding=True, truncation=True, pad_to_max_length=True, return_tensors='np')

np.squeeze(p).shape
```

### Expected behavior

There are several problems:
- It does not return Numpy array so I have to make squeeze operation on my own.
- The bigger problem is that it does not consider padding parameters, the expeced return would be (512, 1024) but now return (6, 1024). I have tried any parameter setup at any level but not worked. When I checked the source code, it only consider truncation parameter.

### PR DESCRIPTION
**Pull request by quancore**
# What does this PR do?

The PR adds keyword arguments for the tokenizer for the feature extraction pipeline. Fixes: https://github.com/huggingface/transformers/issues/19374

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@LysandreJik
@Narsil

### CODE REVIEW

File: src/transformers/pipelines/feature_extraction.py [hunk 1]
```diff
@@ -40,19 +40,17 @@ class FeatureExtractionPipeline(Pipeline):
             the associated CUDA device id.
     """
 
-    def _sanitize_parameters(self, truncation=None, **kwargs):
-        preprocess_params = {}
-        if truncation is not None:
-            preprocess_params["truncation"] = truncation
+    def _sanitize_parameters(self, tokenize_kwargs=None, **kwargs):
```
Comment (Narsil) [inline src/transformers/pipelines/feature_extraction.py:43]: Unfortunately we have to keep `truncation` in there since it was already allowed in order to NOT break anything.

Something like:

```python
def _sanitize_parameters(self, truncation=None, tokenize_kwargs=None, **kwargs):
    # handle tokenize_kwargs first
    if truncation is not None:
         if 'truncation' in tokenize_kwargs:
             raise ValueError("This is defined twice")
         tokenize_kwargs["truncation"] = truncation

**Review by Narsil — COMMENTED** [review]
Thanks for this.

Thanks for the tests too ! 
I added a remark to keep from breaking anything in user code.

**Comment by Narsil** [comment]
After checking, the broken tests are exactly broken by the lack of `truncation` support.

Also for quality you should be able to to 
```
pip install -e .[dev] #  or pip install transformers[dev]
make fixup
```
Cheers.

File: src/transformers/pipelines/feature_extraction.py (hunk 1 above)
Comment (quancore) [reply src/transformers/pipelines/feature_extraction.py:43]: But the problem is truncation is already a parameter of the tokenizer, why the parameter should be kept separately?

**Comment by quancore** [comment]
@Narsil I made the changes you indicate.

File: src/transformers/pipelines/feature_extraction.py (hunk 1 above)
Comment (Narsil) [reply src/transformers/pipelines/feature_extraction.py]: Because it was there before, hence users might (and probably HAVE) started using it.
And since we cannot break user code, we have to keep backward compatibility with it.

**Review by Narsil — APPROVED** [review]
LGTM.

@sgugger

File: tests/pipelines/test_pipelines_feature_extraction.py [hunk 2]
```diff
@@ -52,6 +52,90 @@ def test_small_model_tf(self):
...
+
+        import numpy as np
```
Comment (sgugger) [inline tests/pipelines/test_pipelines_feature_extraction.py:66]: This can go on top, `numpy` is a hard dep of Transformers.

File: tests/pipelines/test_pipelines_feature_extraction.py [hunk 3]
```diff
@@ -52,6 +52,90 @@ def test_small_model_tf(self):
...
```
Comment (sgugger) [inline tests/pipelines/test_pipelines_feature_extraction.py:108]: Same here.

**Review by sgugger — APPROVED** [review]
A couple of nits, but LGTM otherwise! Thanks a lot for working on this!

**Comment by sgugger** [comment]
Thanks a lot!
